File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int p1 = 0, p2 = 0, N = word1.length(), M = word2.length();
        string result = "";
        while (p1 < N && p2 < M) {
            result += word1[p1++];
            result += word2[p2++];
        }
        if (p1 < N)
            result += word1.substr(p1);
        if (p2 < M)
            result += word2.substr(p2);
        return result;
    }
};

File: LeetCode/1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word1_len = len(word1)
        word2_len = len(word2)
        max_length = max(word1_len, word2_len)
        result = ""
        for i in range(max_length):
            if(i < word1_len):
                result += word1[i]
            if(i < word2_len):
                result += word2[i]
        return result
        